src/main/resources/de/itemis/mps/gradle/mpssign.sh

```diff
@@ -1,27 +1,42 @@
 #!/bin/bash
-if [[ $# -ne 3 && $# -ne 2 ]]; then
-  cat <<EOF
-Usage:
+USAGE="Usage:
 
-  $0 RCP_FILE OUTPUT_DIR [JDK_FILE]
+$0 -r <rcp_file> -o <output_dir> [-j <jdk_file>] [-p <password keychain> -k <keychain> -i <sign identity>]
 
 1. Extracts RCP_FILE into OUTPUT_DIR
 2. Creates symlinks Contents/bin/*.dylib -> *.jnilib
-3. If JDK_FILE is given, extracts JDK_FILE under Contents/jre/
+3. If JDK_FILE is given extracts JDK_FILE under Contents/jre/
 4. Builds help indices using hiutil if help is present under Contents/Resources/
 5. Sets executable permissions on Contents/MacOS/* and appropriate Contents/bin/ files
-
+6. If given, signs the application with the passed SIGN_PW, SIGN_KEY_CHAIN and SIGN_IDENTITY
 IMPORTANT: All arguments must use absolute paths because the script changes the current directory several times!
-EOF
-  exit 1
-fi
-
+"
 set -o errexit # Exit immediately on any error
 
-# Arguments
-RCP_FILE="$1"
-OUTPUT_DIR="$2"
-JDK_FILE="$3"
+# Parse arguments
+while getopts ":r:o:j:p:k:i:h" option; 
+do
+  case "${option}" in
+    r) RCP_FILE="$OPTARG";;
+    o) OUTPUT_DIR="$OPTARG";;
+    j) JDK_FILE="$OPTARG";;
+    p) SIGN_PW="$OPTARG";;
+    k) SIGN_KEY_CHAIN="$OPTARG";;
+    i) SIGN_IDENTITY="$OPTARG";;
+    h) echo "$USAGE" 
+       exit 0;;
+    \?) echo "illegal option: -$OPTARG usage: /.mpssing.sh -r <rcp_file> -o <output_dir> [-j <jdk_file>] [-p <password keychain> -k <keychain> -i <sign identity>]" >&2
```

Comment:
Should probably just display $USAGE here. Also, `/.mpssing.sh` is wrong: it should be `./` and "sign", not "sing". Even better, just use $0.

Reply:
little singing script... why not? ))